packages/driver/src/cross-origin/events/socket.ts

```diff
@@ -4,7 +4,15 @@ export const handleSocketEvents = (Cypress) => {
   const webSocket = client({
     path: Cypress.config('socketIoRoute'),
     transports: ['websocket'],
-  }).connect()
+  })
+
+  webSocket.on('connect_error', () => {
+    // fall back to polling if websocket fails to connect (webkit)
+    // https://github.com/socketio/socket.io/discussions/3998#discussioncomment-972316
+    webSocket.io.opts.transports = ['polling', 'websocket']
+  })
```

Comment:
It turns out that this fall-back approach causes issues. On some projects (like `cypress-documentation`), the websocket will take the maximum time to error out (20 seconds, as defined by socket.io), which results in a 20-second delay before the websocket connection gets established. This websocket connection code should be updated to explicitly use `Cypress.isBrowser('webkit')` to avoid the possible 20-second delay. Here's how it's done outside of x-origin:

https://github.com/cypress-io/cypress/blob/26a71fe0193f74758e5f03b141ee76aafa367c17/packages/app/src/runner/index.ts#L34-L42

This could be a good chance to unify these two areas as well, so the implementations don't get further out of sync.

Reply:
Unified here: https://github.com/cypress-io/cypress/pull/23442/commits/d3085db57ebbdb61da73c4dba3ed3999400a6ade

I moved the socket creation logic into the socket package because the driver package cannot depend on the app package.